### ISSUE TITLE
 java.lang.Error: FileProvider not declared or has wrong authority. (Must be ${applicationId}.com.vansuita.pickimage.provider, check AndroidManifest.xml)

### ISSUE THREAD

**Issue by myinnos** [jrvansuita/PickImage#19]

**Comment by jrvansuita** [comment]
Probably you skipped the first implementation section on readme.

You need to copy the exact provides declaration lines to you manifest:
```xml
<manifest ...>
    ... 
    <provider
        android:name="android.support.v4.content.FileProvider"
        android:authorities="${applicationId}.com.vansuita.pickimage.provider"
        tools:replace="android:authorities">
        <meta-data
            android:name="android.support.FILE_PROVIDER_PATHS"
            android:resource="@xml/provider_paths" />
    </provider>
</manifest> 
```

The use of this library can cause INSTALL_FAILED_CONFLICTING_PROVIDER if you skip this step. Update your AndroidManifest.xml with this exact provider declaration below.

**Comment by myinnos** [comment]
I added still same issue!

**Comment by jrvansuita** [comment]
Paste here your manifest please. Also, try to debug it. It's weird this is happening to you. Nobody has ever contacted me because of this.

**Comment by bmbariah** [comment]
Actually it is suppoed to be 

```xml
<manifest ...>
    ... 
<application
    <provider
        android:name="android.support.v4.content.FileProvider"
        android:authorities="${applicationId}.com.vansuita.pickimage.provider"
        tools:replace="android:authorities">
        <meta-data
            android:name="android.support.FILE_PROVIDER_PATHS"
            android:resource="@xml/provider_paths" />
    </provider>
</application
</manifest> 
```

**Comment by jrvansuita** [comment]
@bmbariah I will fix the readme.

**Comment by sayah-y** [comment]
Hello,

I have the same error even if I copy-paste the code in the AndroidManifest file.
I debug and I find this:
<img width="1100" alt="capture d ecran 2017-02-27 a 20 17 16" src="https://cloud.githubusercontent.com/assets/2410393/23375882/46326210-fd2a-11e6-9237-903953b0537c.png">

I am not able to use the library now.

Thanks.

**Comment by myinnos** [comment]
@sayah-y 

This error occurring when it is conflict to other libraries, **for example** `i used this library with out fire-base its works well, but in other project where i used fire-base the same bug happening.`

Please post your library `dependencies` here i will tell you why it is not allowing

**Comment by sayah-y** [comment]
@myinnos thanks for your help.

Here are my lybrary dependencies:
```
// -----------------------------------------
//            GOOGLE SUPPORTS
// -----------------------------------------
compile 'com.android.support:appcompat-v7:25.1.1'
compile 'com.android.support:support-v4:25.1.1'
compile 'com.android.support:design:25.1.1'
// -----------------------------------------
//            NETWORK
// -----------------------------------------
compile('com.jakewharton.retrofit:retrofit2-rxjava2-adapter:1.0.0') {
exclude group: 'io.reactivex.rxjava2'
}
compile 'io.reactivex.rxjava2:rxjava:2.0.1'
compile 'io.reactivex.rxjava2:rxandroid:2.0.1'
compile 'com.trello.rxlifecycle2:rxlifecycle-components:2.0.1'
compile 'com.squareup.retrofit2:converter-gson:2.1.0'
// -----------------------------------------
//            STORAGE
// -----------------------------------------
compile 'io.objectbox:objectbox-android:0.9.7'
compile 'com.orhanobut:hawk:2.0.1'
// -----------------------------------------
//            PICTURE
// -----------------------------------------
compile 'com.github.bumptech.glide:glide:3.7.0'
compile 'com.davemorrissey.labs:subsampling-scale-image-view:3.6.0'
compile 'com.github.jrvansuita:PickImage:v2.0.9'
// -----------------------------------------
//            SIMPLIFY CODE
// -----------------------------------------
compile 'com.jakewharton:butterknife:8.5.1'
annotationProcessor 'com.jakewharton:butterknife-compiler:8.5.1'
// -----------------------------------------
//                UI
// -----------------------------------------
compile 'com.github.dmytrodanylyk.android-process-button:library:1.0.4'
compile 'com.github.JakeWharton:ViewPagerIndicator:2.4.1@aar'
compile 'com.stepstone.stepper:material-stepper:2.2.0'
// -----------------------------------------
//            DEBUG/TEST
// -----------------------------------------
compile 'com.facebook.stetho:stetho-okhttp3:1.4.2'
compile 'com.orhanobut:logger:1.15'
testCompile 'junit:junit:4.12'
androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
exclude group: 'com.android.support', module: 'support-annotations'
exclude group: 'com.google.code.findbugs', module: 'jsr305'
})
```

**Comment by myinnos** [comment]
@sayah-y  you are using so many libraries, We need to test one to one! Do one thing download this repo and change the layout/class names to use. But frankly no idea which class/layout exactly conflicts 

@jrvansuita hope so you got the issue! please use all files/layouts names as unique

**Comment by sayah-y** [comment]
@myinnos thanks for your help.

@jrvansuita I hope that the issue will be fixed.

Thanks.

**Comment by jrvansuita** [comment]
Hi @sayah-y, are you getting "FileProvider not declared or has wrong authority" error?

**Comment by sayah-y** [comment]
Hi @jrvansuita !

My Android Manifest is:
```
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    package="XXXXXXX">

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.CAMERA" />

    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />

    <uses-feature
        android:name="android.hardware.camera.autofocus"
        android:required="false" />
    <uses-feature
        android:name="android.hardware.camera"
        android:required="false" />

    <application
        android:name=".application.App"
        android:allowBackup="true"
        android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
        android:icon="@mipmap/ic_launcher"
        android:label="@string/app_name"
        android:supportsRtl="true"
        android:theme="@style/AppTheme"
        tools:replace="android:configChanges">
        <activity android:name=".ui.activity.AuthActivity">
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />

                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
        </activity>
        <activity android:name=".ui.activity.HomeActivity" />
        <activity android:name=".ui.activity.NCResolveActivity" />
        <activity android:name=".ui.activity.NCSaveActivity" />

        <!-- Image picker -->
        <provider
            android:name="android.support.v4.content.FileProvider"
            android:authorities="${applicationId}.com.vansuita.pickimage.provider"
            tools:replace="android:authorities">
            <meta-data
                android:name="android.support.FILE_PROVIDER_PATHS"
                android:resource="@xml/provider_paths" />
        </provider>
    </application>

</manifest>
```

I upload the screenshot when you can see the exception:
<img width="1100" alt="capture d ecran 2017-02-27 a 20 17 16" src="https://cloud.githubusercontent.com/assets/2410393/23476888/7fb0810a-febc-11e6-91a4-1cdc205741a9.png">

If you don't see the screenshot:
java.lang.IllegalArgumentException: Failed to find configured root that contains /PickResult/current

Thanks for your help.

**Comment by sayah-y** [comment]
It seems that it's an issue with the xml definition...

`
<meta-data
                android:name="android.support.FILE_PROVIDER_PATHS"
                android:resource="@xml/provider_paths" />
`

**Comment by jrvansuita** [comment]
Ok, I got the problem. 

Correct provider declaration:

```xml

       <provider
            android:name="android.support.v4.content.FileProvider"
            android:authorities="${applicationId}.com.vansuita.pickimage.provider"
            android:exported="false"
            android:grantUriPermissions="true"
            tools:replace="android:authorities">
            <meta-data
                android:name="android.support.FILE_PROVIDER_PATHS"
                android:resource="@xml/picker_provider_paths" />
        </provider>`
```


Please, try the last commit and give my a feedback. I won't release a new library version until you get it working.

**Comment by stevyhacker** [comment]
Getting the same issue and I have added the provider tag in manifest properly. Gallery works ok, Camera crashes.

**Comment by stevyhacker** [comment]
@bmbariah soultion fixes the issue, please update the readme.

**Comment by sayah-y** [comment]
Hello @jrvansuita ,

Good work, it's fixed :)

Before the release, you can also update the build.gradle with:
```
compileSdkVersion 25
buildToolsVersion "25.0.2"
```
And
```
compile 'com.android.support:appcompat-v7:25.2.0'
compile 'com.android.support:cardview-v7:25.2.0'
compile 'com.android.support:support-annotations:25.2.0'
```

For the PickResult, you forget to add on the interface IconGravity of PickSetup:
```
@Retention(RetentionPolicy.SOURCE)
```

Thanks again !

**Comment by jrvansuita** [comment]
@stevyhacker The file provider declaration in readme is correct for this new release.
@sayah-y Will fix it on the next release (Current 2.1.0).

**Comment by sayah-y** [comment]
Perfect, thanks !
